zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/api/ProcessingResultBuilder.java

```diff
@@ -87,4 +87,6 @@ ProcessingResultBuilder withResponse(
   ProcessingResult build();
 
   boolean canWriteEventOfLength(int eventLength);
+
+  void setRequestId(long requestId);
```

Comment:
❌ It doesn't make sense to me that this method exists on the `ProcessingResultBuilder` interface. The interface is used by the engine to provide results, but this method is providing details from the stream platform that are not adjusted by the engine. 

Where used in `ProcessingStateMachine`, the code always first creates the concrete implementation `BufferedProcessingResultBuilder` before calling the `setRequestId` method on it. So we can easily remove it from the interface by using `var` rather than the interface name explicitly.

Reply:
so you're suggesting to move it to the concrete implementation instead? I don't mind that.
When I added it to the interface, I also thought I might need to have it mocked somewhere in tests. but so far there's no need.